Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -220,34 +222,30 @@ int CLuaResourceDefs::GetResourceConfig(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    // Verify arguments
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
-
-    if (!argStream.HasErrors())
+    if (!resourceElement)
     {
-        if (pResource)
-        {
-            // Grab its name and return it
-            const char* szName = pResource->GetName();
-            if (szName)
-            {
-                lua_pushstring(luaVM, szName);
-                return 1;
-            }
-        }
-        else
-            m_pScriptDebugging->LogBadPointer(luaVM, "resource", 1);
+        CLuaMain*  localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+        CResource* localResource;
```

Comment:
It's C++, not C89, there's no need in variables declaration before being use. You can declare this varible and initialize it in one shot at line 235.

Reply:
This is my habit, this is my preferred route, but I made the modification, it doesn't matter to me.